src/v/config/configuration.cc

```diff
@@ -2506,6 +2506,28 @@ configuration::configuration()
       "Size of the zstd decompression workspace",
       {.visibility = visibility::tunable},
       8_MiB)
+  , lz4_decompress_buffer_size_bytes(
+      *this,
+      "lz4_decompress_buffer_size_bytes",
+      "Size of buffers preallocated for LZ4 decompression. Two buffers "
+      "allocated per shard, so twice of this value is reserved at startup",
+      {.needs_restart = needs_restart::yes, .visibility = visibility::tunable},
+      4_MiB + 128_KiB)
```

Comment:
> Is there a comment in the code somewhere that explains why this default makes sense?

I'm guessing it comes from something like https://github.com/lz4/lz4/blob/dev/doc/lz4_Block_format.md#large-lengths but i'm not sure about the 128 KB part.

Reply:
See https://github.com/redpanda-data/redpanda/pull/17385#discussion_r1549556794, this number is from the lz4 source code, specifically https://github.com/lz4/lz4/blob/dev/lib/lz4frame.c#L1678-L1679

There are two buffers, a tmpIn and a tmpOut. The tmpIn buffer is at most 4MiB + 4 bytes, the tmpOut buffer is at most 4MiB + 128KiB. I initially thought of using these specific sizes for the two buffers but then used the higher size for both. 

Do note that the extra 128_KiB is allocated only if `dctx->frameInfo.blockMode==LZ4F_blockLinked` and linked block compression is turned off by default for redpanda, but I see no reason not to allocate the extra 128_KiB in case something changes later. 

Unfortunately this code is closely tied with lz4 internals but I did not see any exposed values to drive this setting.